server/Mistakes.Journal.Api/Api/Mistakes/Controllers/MistakesController.cs

```diff
@@ -46,22 +68,65 @@ public async Task<ActionResult<MistakeWebModel>> CloneMistake(Guid mistakeId)
         {
             var mistake = await _dataContext.Set<Mistake>()
                 .Include(m => m.Tips)
+                .Include(m => m.MistakeLabels)
+                .ThenInclude(m => m.Label)
+                .Include(m => m.Repetitions)
                 .SingleOrDefaultAsync(m => m.Id == mistakeId);
 
             if (mistake is null)
                 return NotFound();
 
-            var newMistake = new NewMistakeWebModel
-            {
-                Name = mistake.Name,
-                Goal = mistake.Goal,
-                Priority = mistake.Priority,
-                Tips = mistake.Tips.Select(t => t.Content).ToList(),
-            };
+            var newMistake = mistake.ToNewMistakeWebModel();
 
             return await AddMistake(newMistake);
         }
 
+        [HttpPost("{mistakeId:guid}/add-repetiton")]
+        public async Task<ActionResult<MistakeWebModel>> AddRepetitionResetCounter(Guid mistakeId, [FromBody, MJIncorrectMistakeDate(1)] DateTime? dateTime)
+        {
+            if (!ModelState.IsValid)
+                return BadRequest(ModelState);
+
+            var mistake = await _dataContext.Set<Mistake>()
+                .Include(m => m.Repetitions)
+                .SingleOrDefaultAsync(m => m.Id == mistakeId);
+
+            if (mistake is null)
+                return NotFound();
+
+            mistake.Repetitions.Add(new Repetition(dateTime ?? DateTime.Now));
+
+            _dataContext.Entry(mistake).State = EntityState.Modified;
+            await _dataContext.SaveChangesAsync();
+
+            return Ok(mistake.ToWebModel());
+        }
+
+
+        [HttpPost("search")]
+        public async Task<ActionResult<MistakeWebModel>> GetMistakes(MistakeSearchWebModel searchModel)
+        {
+            var mistakes = await _dataContext.Set<Mistake>()
+                .Include(m => m.Tips)
+                .Include(m => m.Repetitions)
+                .Include(m => m.MistakeLabels)
+                .ThenInclude(m => m.Label)
+                .WhereIf(searchModel.Name.IsPresent(), m => m.Name.ToLower().Contains(searchModel.Name.ToLower()))
+                .WhereIf(searchModel.Goal.IsPresent(), m => m.Goal != null && m.Goal.ToLower().Contains(searchModel.Goal.ToLower()))
+                .WhereIf(!searchModel.Priorities.IsNullOrEmpty(), m => searchModel.Priorities.Contains(m.Priority))
+                .ToListAsync();
+
+            mistakes = mistakes
+                    // this is too hard for .NET Core
```

Comment:
Damn, I know, but we use pagination :/ With that it is not so useful. Why is it that hardcore? This label search makes it that way?